sdk/python/feast/infra/offline_stores/redshift.py

```diff
@@ -58,3 +100,61 @@ def get_historical_features(
         project: str,
     ) -> RetrievalJob:
         pass
+
+
+class RedshiftRetrievalJob(RetrievalJob):
+    def __init__(self, query: str, redshift_client, s3_resource, config: RepoConfig):
```

Comment:
Can we please add type annotations and maybe a docstring to `__init__`? It's not clear to me what `s3_resource` is.

Reply:
we can't add type annotations to boto3 client & resource objects, because of how boto3 generates these objects. I have type annotations for the rest of the objects.

Reply:
I don't understand. So there is no class `s3_resource` associated class or type?

Reply:
Yes. Check out
https://stackoverflow.com/questions/52087307/adding-type-hinting-to-functions-that-return-boto3-objects
https://stackoverflow.com/questions/51118767/what-is-boto3-clients3-returning